### PR TITLE
[ML] Migrate to Backstage for BuildKite pipeline definitions

### PR DESCRIPTION
**Pull request by edsavage**
Add a "catalog-info.yaml" file that contains definitions of the ml-cpp BuildKite pipelines. These will take precedence of those defined in the CI repo. Once the catalog is merged to main and detected by the CI Backstage system the old ml-cpp* pipeline definition files in the CI repo can be deleted.

Also note that the specification of the pipeline files has been changed to refer to the python scripts directly. This is taking advantage of the fact that the CI infrastructure will run such executable scripts if detected.

### CODE REVIEW

**Comment by edsavage** [comment]
retest

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        publish_commit_status: false
+        publish_commit_status_per_step: true
+        trigger_mode: code
+      repository: elastic/ml-cpp
+      schedules:
+        Daily 7_17:
+          branch: '7.17'
+          cronline: 30 03 * * *
+          message: Daily SNAPSHOT build for 7.17 
+        Daily 8_7:
```
Comment (droberts195) [inline catalog-info.yaml:169]: 8.7 can be deleted now 8.8 is released.

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by edsavage** [comment]
Merging this now, without waiting for Jenkins CI, as these changes by their nature do not relate to Jenkins builds

**Pull request merged**